float64[] sVal
uint16 size